### PR TITLE
Added Slack integration

### PR DESCRIPTION
**Pull request by dymissy**
This PR enables you to send message to your Slack channels though Slack APIs.

### CODE REVIEW

File: idxfile.php.dist [hunk 1]
```diff
@@ -145,4 +175,5 @@ $idx->
 
 $idx->addLibrary('deploy', new Deploy());
 $idx->addLibrary('phpunit', new PHPUnit());
+$idx->addLibrary('slack', new Slack($slackParams));
```
Comment (micheleorselli) [inline idxfile.php.dist]: Is this the way we want do to this? Maybe is Idephix responsibility to inject parameters in the extension, a la Symfony2, 
What do you think guys?

File: src/Idephix/Extension/Slack/Slack.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+class Slack implements IdephixAwareInterface
+{
+    /**
+     * @var \Idephix\IdephixInterface
+     */
+    private $idx;
+
+    private $settings;
+
+    public function __construct($args = array())
```
Comment (micheleorselli) [inline src/Idephix/Extension/Slack/Slack.php]: in other extensions (see https://github.com/ideatosrl/Idephix/blob/master/src/Idephix/Extension/Project/Project.php) parameters are passed via retrieved via $idx. I think we should maintain consistency

Comment (ricfrank) [inline src/Idephix/Extension/Slack/Slack.php]: :+1: 

Consistency is important to maintain.

@micheleorselli: we could remove the constructor of the class and inject all needed parameters to ```sendToSlack```method. Is this your suggestion ?

File: idxfile.php.dist (hunk 1 above)
Comment (ricfrank) [inline idxfile.php.dist]: @micheleorselli is it related to the comment below ?

File: src/Idephix/Extension/Slack/Slack.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        $data = "payload=" . json_encode(array(
+                "channel" => $channel,
+                "text" => $message,
+                "icon_url" => $icon_url,
+                "unfurl_links" => true,
+                "username" => $username,
+                "attachments" => $attachments
+            ));
+
+        $result = $this->request($data, $this->settings['url']);
```
Comment (ricfrank) [inline src/Idephix/Extension/Slack/Slack.php]: What do you think to change the name of ```request()```method ? It isn't a verb and imo methods should be verb.

Do you like a name as ```send``` or ```doRequest``` ? For example Guzzle use ```send``` method.

In terms of readability

```$response = $this->send($data, $url)```

is better than

```$result = $this->request($data, $this->settings['url']);````

File: idxfile.php.dist (hunk 1 above)
Comment (micheleorselli) [inline idxfile.php.dist]: yup :-)

File: src/Idephix/Extension/Slack/Slack.php (hunk 2 above)
Comment (micheleorselli) [inline src/Idephix/Extension/Slack/Slack.php]: well yes, inside the sendToSlack method, using $idx->get('param') the extension can get every parameter it needs

File: src/Idephix/Extension/Slack/Slack.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            'channel' => '#general',
+            'icon_url' => 'https://slack.com/img/icons/app-57.png',
+            'username' => 'slackbot'
+        );
+
+        $this->settings = array_merge($defaults, $args);
+    }
+
+    public function sendToSlack($message, $attachments = array(), $channel = '', $icon_url = '', $username = '')
+    {
```
Comment (micheleorselli) [inline src/Idephix/Extension/Slack/Slack.php]: Are we sure we want to expose all these parameters ? The final user can set all these data in the ```$config``` array
What the ```$attachment``` parameter is supposed to contain? In the ```idxfile.php.dist``` we have 

```php
array(
 array(
  'fallback' => "Hi guys, I'm Idephix and this is my first slack message!",
  'pretest'  => "Hi guys, I'm Idephix and this is my first slack message!",
  'color'    => "#43e17b",
  'fields'   => array(
  array(
    'title' => 'I\'m Idephix',
    'value' => "Hello world, aren't you using Idephix?\nWhy not? Please <http://getidephix.com|download Idephix> right now",
   'short' => false
      )
    )
) 
```

Moreover but I see a couple more issues in the $attachment param:
1. $attachment  is perhaps the wrong name
2. the data passed is extremely complicated
3. how the final user is supposed to know the right structure to pass?

File: src/Idephix/Extension/Slack/Slack.php (hunk 2 above)
Comment (dymissy) [inline src/Idephix/Extension/Slack/Slack.php]: :+1: got it!

File: src/Idephix/Extension/Slack/Slack.php (hunk 4 above)
Comment (dymissy) [inline src/Idephix/Extension/Slack/Slack.php]: >Are we sure we want to expose all these parameters ? The final user can set all these data in the $config array.

`$attachments` is optional. Only the `message` parameter is required. I've added the full array just as example. `$config` array could be a solution but what if I want to customize different kind of messages?

>What the $attachment parameter is supposed to contain?

`$attachments` contains parameters supported from APIs. 

>1. $attachment is perhaps the wrong name
>2. the data passed is extremely complicated
>3. how the final user is supposed to know the right structure to pass?

1. attachment is the name that Slack uses in its own APIs. I just mantained the name.
2. I'm totally agree with you but I've just used the parameters requested from Slack APIs. Suggestions in order to make it easier?
3. I've just attached Slack documentation links in the annotation of the task and I thought to add a section in Idephix documentation when the PR will be approved.

File: src/Idephix/Extension/Slack/Slack.php (hunk 3 above)
Comment (dymissy) [inline src/Idephix/Extension/Slack/Slack.php]: Done!

**Comment by ricfrank** [comment]
@dymissy @micheleorselli what about this PR ?

**Pull request merged**